generators/app/index.js

```diff
@@ -178,29 +179,51 @@ function handleAspNet(yo) {
     var aspNet = new AspNetHelper(baseImageName, portNumber);
 
     var done = yo.async();
-    aspNet.configureUrls(function (err, noteForUser) {
+    aspNet.updateProjectJson(function (err, projectJsonNote) {
         if (err) {
             error = true;
             yo.log.error(err);
+            done();
             return;
         }
-        configureUrlsNoteForUser = noteForUser;
-        done();
+        updateProjectJsonNoteForUser = projectJsonNote;
+
+        if (aspNet.getDotnetVersion() == 'RC2') {
+            aspNet.updateProgramCS(function (err, programCSNode) {
```

Comment:
They are being process separately. updateProgramCS is being called as part of the continuation of calling updateProjectJson. You can see a similar pattern of continuing in the call-back in both updateProjectJson and updateProgramCS's implementations

---
In reply to: [67465190](https://github.com/Microsoft/generator-docker/pull/69#discussion_r67465190) [](ancestors = 67465190)